modin/core/storage_formats/pandas/groupby.py

```diff
@@ -0,0 +1,213 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Contains implementations for GroupbyReduce functions."""
```

Comment:
It seems to me that is a good point to create kernels folder and put the content there like `storage_formats/pandas/kernels/groupby.py`. What do you think?

Reply:
I don't think we should do it this way.

When I'm thinking about kernels I think about some unified approach across all of the QCs. I imagine this as some kind of accessor defined for every kernel inside the base QC:
```python
class BaseQueryCompiler:
    ...
    @property
    def groupby(self):
        return BaseQCGroupby(self, **maybe_some_other_params)
    ...

# query_compiler.py
# kernels/
#     groupby.py -> class BaseQCGroupby(maybe it could even inherit from QC? BaseQueryCompiler): ...
#     ....

def API_level_code():
    self._query_compiler.groupby.mean()
    self._query_compiler.rolling.sum()
    self._query_compiler.str.split()
```

What I did in this PR is just created a helper class that just moved the code out from PandasQueryCompiler without touching & resolving the actual problem of excessive QC API. Don't think it is worth naming this a "kernel".

Reply:
> When I'm thinking about kernels I think about some unified approach across all of the QCs. I imagine this as some kind of accessor defined for every kernel inside the base QC

IIUC your thoughts, this can't be unified approach across all of the QCs (everything in BaseQC). Each single QC must have knowledge of the concrete compute kernels and concrete in-memory format of the data.

Ok, let's leave the change for the original issue regarding "kernels" concept.